tests/e2e.rs

```diff
@@ -1,30 +1,94 @@
 use assert_cmd::prelude::*;
 use regex::Regex;
 use std::fs;
-use std::process::Command;
+use std::process::{Child, Command};
 use std::thread;
 use std::time;
 
 mod utils;
 
+// ayb_cmd!("value1", value2; {
+//     "ENV_VAR" => env_value
+// })
+macro_rules! ayb_cmd {
+    ($($value:expr),+; { $($env_left:literal => $env_right:expr),* $(,)? }) => {
+        Command::cargo_bin("ayb")?
+                .args([$($value,)*])
+                $(.env($env_left, $env_right))*
+    }
+}
+
+// ayb_assert_cmd!("value1", value2; {
+//     "ENV_VAR" => env_value
+// })
+macro_rules! ayb_assert_cmd {
+    ($($value:expr),+; { $($env_left:literal => $env_right:expr),* $(,)? }) => {
+        Command::cargo_bin("ayb")?
+                .args([$($value,)*])
+                $(.env($env_left, $env_right))*
+                .assert()
+                .success()
+    }
+}
+
+struct Cleanup;
+
+impl Drop for Cleanup {
+    fn drop(&mut self) {
+        if let Err(err) = fs::remove_dir_all("/tmp/ayb/e2e") {
+            assert_eq!(format!("{}", err), "No such file or directory (os error 2)")
+        }
+    }
+}
+
+struct AybServer(Child);
+impl AybServer {
+    fn run(db_type: &str) -> Result<Self, Box<dyn std::error::Error>> {
+        let child =
```

Comment:
For consistency with `SmtpServer`, should we wrap this in an `Ok(...)` without declaring `let child`?

Reply:
That doesn't compile, a let binding is needed there

Reply:
Done!